modules/dnn/test/test_graph_simplifier.cpp

```diff
@@ -0,0 +1,137 @@
+    test("two_resizes_with_shared_subgraphs", std::vector<std::string>{"NaryEltwise", "Resize"});
+}
+
+TEST_P(Test_Graph_Simplifier, SoftmaxSubgraph) {
+    /* Test for 3 subgraphs
+        - SoftMaxSubgraph
+        - SoftMaxSubgraph2 (conformance)
+        - LogSoftMaxSubgraph (conformance)
+    */
+    test("softmax_unfused", "Softmax");
+    test_conformance("test_softmax_example_expanded", "Softmax");
+    test_conformance("test_softmax_axis_2_expanded", "Softmax");
+    test_conformance("test_softmax_default_axis_expanded", "Softmax");
+    test_conformance("test_softmax_axis_0_expanded", "Softmax");
+    test_conformance("test_softmax_axis_1_expanded", "Softmax");
+    test_conformance("test_softmax_large_number_expanded", "Softmax");
+    test_conformance("test_softmax_negative_axis_expanded", "Softmax");
+    test_conformance("test_logsoftmax_axis_2_expanded", "Softmax");
+    test_conformance("test_logsoftmax_example_1_expanded", "Softmax");
+    test_conformance("test_logsoftmax_negative_axis_expanded", "Softmax");
+    test_conformance("test_logsoftmax_axis_0_expanded", "Softmax");
+    test_conformance("test_logsoftmax_axis_1_expanded", "Softmax");
+    test_conformance("test_logsoftmax_large_number_expanded", "Softmax");
+    test_conformance("test_logsoftmax_default_axis_expanded", "Softmax");
+}
+
+TEST_P(Test_Graph_Simplifier, HardSwishSubgraph) {
+    test_conformance("test_hardswish_expanded", "HardSwish");
+}
+
+TEST_P(Test_Graph_Simplifier, CeluSubgraph) {
+    test_conformance("test_celu_expanded", "Celu");
+}
+
+TEST_P(Test_Graph_Simplifier, NormalizeSubgraph) {
+    /* Test for 6 subgraphs
+        - NormalizeSubgraph1
+        - NormalizeSubgraph2
+        - NormalizeSubgraph2_2
+        - NormalizeSubgraph3
+        - NormalizeSubgraph4
+        - NormalizeSubgraph5
+    */
+    test("reduceL2_subgraph_2", "Normalize");
+    test("reduceL2_subgraph", "Normalize");
+    test("normalize_fusion", "Normalize");
+}
+
+TEST_P(Test_Graph_Simplifier, BatchNormalizationSubgraph) {
+    /* Test for 2 subgraphs
+        - BatchNormalizationSubgraph1
+        - BatchNormalizationSubgraph2
+    */
+    test("frozenBatchNorm2d", "BatchNorm");
+    test("batch_norm_subgraph", "BatchNorm");
+}
+
+TEST_P(Test_Graph_Simplifier, ExpandSubgraph) {
+    test("expand_neg_batch", "Expand");
+}
+
+TEST_P(Test_Graph_Simplifier, MishSubgraph) {
+    /* Test for 2 subgraphs
+        - SoftplusSubgraph
+        - MishSubgraph
+    */
+    test("mish_no_softplus", "Mish");
+    test("mish", "Mish");
+}
+
+// Different backends are sharing the same subgraph fusion rule,
+// so testing on tuple(DNN_BACKEND_OPENCV, DNN_TARGET_CPU) is enough
+using BackendTargetTuple = tuple<Backend, Target>;
+INSTANTIATE_TEST_CASE_P(/*nothing*/, Test_Graph_Simplifier,
+                        testing::ValuesIn(std::vector<BackendTargetTuple>{std::make_tuple(DNN_BACKEND_OPENCV, DNN_TARGET_CPU)}));
```

Comment:
Why not just make all the tests non-parametric?
```
TEST(Test_Graph_Simplifier, ExpandSubgraph) {
}

TEST(Test_Graph_Simplifier, MishSubgraph) {
}
```

Reply:
Thank you! Done with your code.